### PR TITLE
Insert release dates into RELEASE.rst

### LINKED ISSUE #158: write a date to the release notes
Now that we're doing so many releases, it can be hard to trace back a change in app behavior to a specific release. It would be nice to have (human-readable) dates in the release notes to help. 

Ideally we'd record the date and time of the production release. Could we update the release notes at the time the release is merged back into master?

As a one-time activity, it would be nice to retroactively record the dates of each release's tag as well.

### PR DESCRIPTION
**Pull request by mbertrand**
#### What are the relevant tickets?
- Closes #158

#### What's this PR do?
Goes through RELEASE.rst line by line and appends a release date (based on date of version tag) to each line that is equal to "Version &lt;number&gt;", then commits and pushes the change.

#### How should this be manually tested?
- Partial test: In a python shell opened in a repo directory, copy over the `set_release_date` function and run it.
- Run a release

### CODE REVIEW

**Comment by noisecapella** [comment]
It looks like the PR adds an extra commit to change the release date. Instead can we do this when the release notes are written out in `update_release_notes`?

File: finish_release.py [hunk 1]
```diff
@@ -40,6 +44,39 @@ def tag_release(version):
...
+            for line in existing_note_lines:
+                if line.startswith("Version ") and "Released" not in line:
+                    version_match = re.search(r"[0-9\.]+", line)
+                    if version_match:
+                        version_line = version_match.group(0)
+                        version_date = check_output(
+                            ["git", "log", "-1", "--format=%ai", "v{}".format(version_line)]
+                        ).rstrip()
+                        localtime = datetime.strptime(version_date.decode("utf-8"), "%Y-%m-%d %H:%M:%S %z").\
+                            astimezone(pytz.timezone("America/New_York")).strftime(date_format)
```
Comment (noisecapella) [inline finish_release.py]: It's pedantic but can we use `pytz.timezone(envs['TIMEZONE'])` instead?

File: finish_release.py [hunk 2]
```diff
@@ -40,6 +44,39 @@ def tag_release(version):
...
+                        line = "Version {} (Released {})\n".format(version_line, localtime)
+                f.write(line)
+    except:  # pylint:disable=bare-except
+        check_call(["git", "checkout", "HEAD", "--", release_filename])
```
Comment (noisecapella) [inline finish_release.py]: We are always working in a temporary directory so we don't need to clean up on exceptions here. Once the exception raises up past `init_working_dir` the directory will automatically get deleted

**Comment by mbertrand** [comment]
@noisecapella I didn't do this in `update_release_notes` because @pdpinch said it would be best to set the date for the version to when it was actually released to production, whereas `update_release_notes` is run when it is deployed to RC and before a version tag is generated (the tag date is used as the release date).  But this wouldn't be a problem for previous releases, so could just leave the current version as is (with no date) and it would be filled in on the next release; and if we needed to know the date before that we can just check the latest commit date to the release branch?

File: finish_release.py (hunk 2 above)
Comment (mbertrand) [reply finish_release.py]: Removing try..except block

File: finish_release.py (hunk 1 above)
Comment (mbertrand) [reply finish_release.py]: Replacing with `pytz.timezone(getenvs()['TIMEZONE'])`

File: finish_release.py [hunk 3]
```diff
@@ -40,6 +44,39 @@ def tag_release(version):
     check_call(['git', 'push', '--follow-tags'])
 
 
+def set_release_date(version):
+    """Sets the release date(s) in RELEASE.rst for any versions missing it"""
+    print("Setting release date...")
+    release_filename = "RELEASE.rst"
+    if not os.path.isfile(release_filename):
+        return
+    date_format = "%Y-%m-%d"
```
Comment (pdpinch) [inline finish_release.py]: Can we get the date formatted as "July 12, 2018" -- I think that's "%b %d, %Y"

Comment (mbertrand) [reply finish_release.py]: ✅

**Comment by mbertrand** [comment]
@pdpinch any preference on how to handle https://github.com/mitodl/release-script/pull/159#issuecomment-404567308 ?

**Pull request merged**